server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/formula/counter/RatingVariationValueArrayTest.java

```diff
@@ -0,0 +1,129 @@
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.task.projectanalysis.formula.counter;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Optional;
+import org.assertj.guava.api.Assertions;
+import org.junit.Test;
+import org.sonar.server.computation.task.projectanalysis.measure.MeasureVariations;
+import org.sonar.server.computation.task.projectanalysis.period.Period;
+import org.sonar.server.computation.task.projectanalysis.period.PeriodsHolder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.server.computation.task.projectanalysis.qualitymodel.RatingGrid.Rating;
+import static org.sonar.server.computation.task.projectanalysis.qualitymodel.RatingGrid.Rating.A;
+import static org.sonar.server.computation.task.projectanalysis.qualitymodel.RatingGrid.Rating.B;
+import static org.sonar.server.computation.task.projectanalysis.qualitymodel.RatingGrid.Rating.C;
+import static org.sonar.server.computation.task.projectanalysis.qualitymodel.RatingGrid.Rating.D;
+import static org.sonar.server.computation.task.projectanalysis.qualitymodel.RatingGrid.Rating.E;
+
+public class RatingVariationValueArrayTest {
+
+  private static final List<Period> PERIODS;
+
+  static {
+    ImmutableList.Builder<Period> builder = ImmutableList.builder();
+    for (int i = 1; i <= PeriodsHolder.MAX_NUMBER_OF_PERIODS; i++) {
+      builder.add(createPeriod(i));
+    }
+    PERIODS = builder.build();
+  }
+
+  @Test
+  public void newArray_always_returns_a_new_instance() {
+    assertThat(RatingVariationValue.newArray()).isNotSameAs(RatingVariationValue.newArray());
+  }
+
+  @Test
+  public void get_returns_unset_value_for_each_period_index() {
+    RatingVariationValue.Array array = RatingVariationValue.newArray();
+    for (Period period : PERIODS) {
+      RatingVariationValue value = array.get(period);
+      verifyUnsetVariation(value);
+    }
+  }
+
+  @Test
+  public void get_returns_set_value_for_each_period_index_if_increment_has_been_called() {
+    RatingVariationValue.Array array = RatingVariationValue.newArray();
+    for (Period period : PERIODS) {
```

Comment:
It's because this class is a copy of other XXXVariationValue => see them in the same package